src/OrbitQt/main.cpp

```diff
@@ -74,8 +74,28 @@ using Context = orbit_ssh::Context;
 
 Q_DECLARE_METATYPE(std::error_code);
 
+void FilterCommandLineArgs(const std::vector<std::string>& command_line_args,
```

Comment:
I think its a better idea if this is not done by hand, but handled by abseil. 
the call absl::ParseCommandLine returns a vector of arguments that are not flags.

Reply:
Thanks so much for the suggestion.

The purpose of this function is to extract commandline flags and capture file paths from the command line arguments.
I didn't find a good way to using abseil to handle this. The return value of `absl::ParseCommandLine(argc, argv)` will ignore all arguments passed after the flag-terminating delimiter (--), see the [document](https://abseil.io/docs/cpp/guides/flags#parsing-flags-during-startup) and the [test cases](https://git.xkool.org/hw/abseil-cpp/-/blob/2eba343b51e0923cd3fb919a6abd6120590fc059/absl/flags/parse_test.cc#L415). 
Hence, if we start with command line flags like `file_name  --flag_name  flag_value` , then the return value includes both `file_name` and `flag_value`.

Therefore, we cannot get the capture file paths from the return value of `absl::ParseCommandLine(argc, argv)`.

Reply:
I just investigated this and here is my best take (the result is: we should still use the output of `absl::ParseCommandLine()`)

`--flag_name flag_value` actually does not seem to work in general. 
I observed the following:
`./Orbit --devmode true` and `./Orbit --devmode false` both turn on devmode. 
`./Orbit --devmode=true` and `./Orbit --devmode` turns on devmode, `./Orbit --devmode=false` turns it off.

Reply:
I observed the following:
```
> Executing task: ./Orbit --devmode false --sampling_rate 500 <

[2021-02-05T15:24:10.425171] [                    OrbitQt/main.cpp:265] length 2
[2021-02-05T15:24:10.425212] [                    OrbitQt/main.cpp:267] arg: ./Orbit
[2021-02-05T15:24:10.425219] [                    OrbitQt/main.cpp:267] arg: false
```
this turns ON devmode and sets the sampling_rate to 500
So it seems to work for non boolean values. The 500 is also not part of the std::vector returned by absl::ParseCommandLine

Reply:
> I just investigated this and here is my best take (the result is: we should still use the output of `absl::ParseCommandLine()`)
> 
> `--flag_name flag_value` actually does not seem to work in general.
> I observed the following:
> `./Orbit --devmode true` and `./Orbit --devmode false` both turn on devmode.
> `./Orbit --devmode=true` and `./Orbit --devmode` turns on devmode, `./Orbit --devmode=false` turns it off.

Thanks so much for looking into this 👍 

If  we declare devmode as `ABSL_FLAG(bool, devmode, false, "")`, then `--devmode false` turns the devmode off.

If in the future, we add more flags and use `--flag_name flag_value` intentionally or unintentionally, then we will open an Orbit instance with a `flag_value`.
And we still need to get the command line flags by hand. (Not sure about this but I can not find a abseil method to get all the command line flags.)

Reply:
May be we use the return value of `absl::ParseCommandLine()` and check whether the elements included in the return value are paths.

Reply:
> If we declare devmode as ABSL_FLAG(bool, devmode, false, ""), then --devmode false turns the devmode off.

I dont think this is true. I think it turns devmode on and absl::ParseCommandLine returns ["false"]. I tested every way I thought of. Linux & Windows.

Reply:
Maybe we should talk about this offline